dvf_csv/src/Plugin/Visualisation/Source/CsvFile.php

```diff
@@ -3,23 +3,28 @@
 namespace Drupal\dvf_csv\Plugin\Visualisation\Source;
 
 use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
 use Drupal\dvf\Plugin\VisualisationInterface;
 use Drupal\dvf\Plugin\Visualisation\Source\VisualisationSourceBase;
 use GuzzleHttp\Client;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
- * Plugin implementation of the 'dvf_csv_remote' visualisation source.
+ * Plugin implementation of the 'dvf_csv_file' visualisation source.
  *
  * @VisualisationSource(
- *   id = "dvf_csv_remote",
+ *   id = "dvf_csv_file",
```

Comment:
Yes, that is right.

> The main thing I would like to see is more documentation, possibly something added to the readme regarding what will break when updating to this version (and how to fix)

We are still in dev and things are likely to change and break at this point. I don't think we need to document in the module what breaks when updating to this version (what version?) or how to fix. We are still working towards the first beta release mate.

Having said that if you need help writing an upgrade path let me know and we will have a chat about it.

Reply:
Yeah I guess there are only 2 reported sites using - one of them mine :) probably not required and can fumble through the upgrade. 

Possibly something to consider for the future is a UPGRADE.TXT or something which just says what to look for rather than a dev having to debug for a few hours until a fix is found. That said, hopefully, it will be solid enough API that will not be required.